packages/dcos-integration-test/extra/conftest.py

```diff
@@ -162,12 +162,16 @@ def _dump_diagnostics(request, dcos_api_session):
 
         last_datapoint = {
             'time': None,
+            'start': None,
```

Comment:
Mmm... this pattern of passing a `last_datapoint` dictionary is a mess but it is inherited and it makes sense.

It tells us "what was the time and value of the last datapoint observed" - at the start there is no time.

What does it mean to have `last_datapoint['start']`?

Can we instead have a variable here - `time_before_waiting = datetime.datetime.now()`, and not change `wait_for_diagnostics_job`?

Reply:
Separately and not for now - should be be using monotonic time rather than `datetime.datetime.now()`?

Reply:
Start just tracks the start time of the bundle gathering. I considered giving it an initial value of now but decided against it, I think because it would keep all datetime grabbing in one place, but I'm not strongly attached to that.

The idea was that we could then use it to see how long a bundle takes to generate since I don't think we have any idea how long it takes normally especially because this test has a 0% success rate for all the runs it's still tracking so we know that it's more than 2 minutes.

From a practical standpoint though, I'm not sure if anyone will really look at that number because they'll have no reason to unless the bundle fails to generate and if that happens nothing is logged and even if we did it would just log the roughly the length of the timeout.

I wouldn't be against making this PR only the timeout bump.

Reply:
To be clear - I think that this variable should exist but not in the `last_datapoint` dictionary - it can be its own variable and it can be set in this function.